utils.py

```diff
@@ -111,30 +116,77 @@ def make_confusion_matrix(y_true, y_pred, classes=None, figsize=(10, 10), text_s
                 size=text_size,
             )
     plt.show()
+    # path = str()
+    # for name in classes:
+    #     path += name.replace(' ', '-').lower()
+    # path += '.png'
+    # plt.savefig(f"plots/{path}")
 
 
 def test_model(model: nn.Module, test_data: DataLoader):
     """
-    Test the performance of the trained ComposerClassifier model with the provided test data loader
-    .by plotting a confusion matrix.
+    Creates a shuffled copy of the input dataset in as a dict, with additional 'pred' column for model predictions.
 
     Args:
         model (PitchSeqNN): The trained ComposerClassifier model to be evaluated.
         test_data (DataLoader): Data for the model to be tested on
     Returns:
-        Tupre[int, int]: Tuple of (true_labels, predicted_labels)
+        dict: all sample data updated with "pred" key.
     """
     # containers for predictions and truths:
     predicted = torch.tensor([])
     true = torch.tensor([])
+    data = {}
     with torch.no_grad():
-        for data in test_data:
-            notes, labels = data
-            out = model(notes)
+        for batch in test_data:
+            labels = batch["label"]
+            out = model(batch["data"])
             preds = out.argmax(1)
-
+            all_info = batch.copy()
+            # updating sample data with predictions
+            all_info.update({"pred": preds})
+
+            # reshaping notes so that there is one list of notes for each sample instead of batch
+            notes = [
+                {key: [lst[i].item() for lst in lsts] for key, lsts in batch["notes"].items()} for i in range(len(batch["data"]))
+            ]
+            all_info["notes"] = notes
+
+            # merging dictionaries to store all the data in one dict
+            data = merge_dictionary(data, all_info)
             predicted = torch.concatenate((predicted, preds))
             true = torch.concatenate((true, labels))
 
-    # plot a confusion matrix with all the predictions from test data
-    return true, predicted
+    return data
+
+
+def piece_av_files(piece: MidiPiece) -> dict:
+    # stolen from Tomek
+    midi_file = os.path.basename(piece.source["midi_filename"])
+    mp3_path = midi_file.replace(".midi", ".mp3")
+    mp3_path = os.path.join("tmp", mp3_path)
```

Comment:
Make sure this folder exists - for "tmp" I'd say it's the best to include it in the repo by commiting `tmp/.keep` file and addint `tmp/` to .gitignoe.